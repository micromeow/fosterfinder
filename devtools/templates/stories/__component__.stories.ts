import type { Meta, StoryObj } from '@storybook/react';
// @ts-ignore
import { __component__ } from '@/components/ui/__component__';

const meta: Meta<typeof __component__> = {
  title: 'Example/__component__',
  component: __component__,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof __component__>;

export const Primary: Story = {
  args: {},
};
