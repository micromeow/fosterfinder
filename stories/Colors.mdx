{/* Colors.mdx */}

import { Meta, ColorPalette, ColorItem } from "@storybook/blocks";

<Meta label="Colors" />

<h1>Colors</h1>

<ColorPalette>
  <ColorItem
    title="theme.color.greyscale"
    subtitle="Some of the greys"
    colors={{ White: "#FFFFFF", Alabaster: "#F8F8F8", Concrete: "#F3F3F3" }}
  />
  <ColorItem
    title="theme.color.positive"
    subtitle="Green"
    colors={{
      Apple: "rgba(102,191,60,1)",
      Apple80: "rgba(102,191,60,.8)",
      Apple60: "rgba(102,191,60,.6)",
      Apple30: "rgba(102,191,60,.3)",
    }}
  />
</ColorPalette>
