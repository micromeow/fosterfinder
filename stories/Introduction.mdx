{/* Introduction.mdx */}

import { Meta, Source } from "@storybook/blocks";

<Meta label="Introduction" />

<h1>Introduction</h1>
<h2>READ ME BEFORE DOING ANYTHING</h2>
<p>
  This is the introduction to the component library for Foster Finder. It is built on
  the <a href="https://ui.shadcn.com/">shadcn/ui</a> component library, an open
  sourced component library built on{" "}
  <a href="https://tailwindcss.com">tailwindcss</a>.
</p>

<h2>Installing New Components from `shadcn`</h2>
<p>
  shadcn uses a "need-to-have" approach to adding new components. If you need to
  use a component that has not been installed in the `components/ui` directory,
  follow these steps to install it with the correct configuration.
</p>

1 - Install the component using the following command:

<Source code={`npx shadcn-ui@latest add <component-name>`} />

2 - <b>IMPORTANT</b>: Rename the generated component to use PascalCase. For example, if you are adding the component `button`, rename the component to `Button`.

3 - Generate stories and tests for the component using the following command:

<Source code={`npm run codegen`} />
