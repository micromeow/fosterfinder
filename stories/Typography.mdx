{/* Typography.mdx */}

import { Meta } from "@storybook/blocks";

<Meta label="Typography" />

<h1>Typography</h1>

<a href="https://ui.shadcn.com/docs/components/typography">
  https://ui.shadcn.com/docs/components/typography
</a>
