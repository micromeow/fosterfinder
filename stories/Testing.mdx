{/* Introduction.mdx */}

import { Meta, Source } from "@storybook/blocks";

<Meta label="Testing" />

<h1>Testing</h1>

Each component should at minimum have a snapshot test. This ensures that the
component is rendered correctly and that it doesn't change unexpectedly. To
run the tests, run the following command:

<Source code={`npm test`} />

<h2>Creating a test</h2>

You may use the following command to generate a test and storybook file for a component:

<Source code={`npm run codegen`} />

Or you may do it manually:

<Source code={`
import * as stories from "./Button.stories";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/react";

const asyncButtonStories = composeStories(stories);

const testCases = Object.values(asyncButtonStories).map((Story) => [
	// The ! is necessary in Typescript only, as the property is part of a partial type
	Story.storyName!,
	Story,
]);

test.each(testCases)("Renders %s story", async (\_storyName, Story) => {
	const tree = await render(<Story />);
	expect(tree.baseElement).toMatchSnapshot();
});
`}/>
