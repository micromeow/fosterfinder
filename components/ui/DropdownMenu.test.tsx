import * as stories from './DropdownMenu.stories';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

const DropdownMenuStories = composeStories(stories);

const testCases = Object.values(DropdownMenuStories).map((Story) => [
  // The ! is necessary in Typescript only, as the property is part of a partial type
  Story.storyName!,
  Story,
]);

// Skipped due to unresolved test error, t resolve in another pr
test.skip.each(testCases)('Renders %s story', async (_storyName, Story) => {
  const tree = await render(<Story />);
  expect(tree.baseElement).toMatchSnapshot();
});
