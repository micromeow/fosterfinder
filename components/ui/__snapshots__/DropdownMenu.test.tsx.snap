// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Primary story 1`] = `
<body
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r1:"
      aria-expanded="true"
      aria-haspopup="menu"
      class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:-translate-y-0.5 transition border border-input bg-background h-10 px-4 py-2"
      data-state="open"
      id="radix-:r0:"
      type="button"
    >
      Open by default!
    </button>
  </div>
  <div
    data-radix-popper-content-wrapper=""
    dir="ltr"
    style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content;"
  >
    <div
      aria-labelledby="radix-:r0:"
      aria-orientation="vertical"
      class="z-50 min-w-[8rem] overflow-hidden rounded-md border bg-popover p-1 text-popover-foreground shadow-md data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 w-56"
      data-align="center"
      data-orientation="vertical"
      data-radix-menu-content=""
      data-side="bottom"
      data-state="open"
      dir="ltr"
      id="radix-:r1:"
      role="menu"
      style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height); animation: none; pointer-events: auto;"
      tabindex="-1"
    >
      <div
        class="px-2 py-1.5 text-sm font-semibold"
      >
        My Account
      </div>
      <div
        aria-orientation="horizontal"
        class="-mx-1 my-1 h-px bg-muted"
        role="separator"
      />
      <div
        role="group"
      >
        <div
          class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
          data-orientation="vertical"
          data-radix-collection-item=""
          role="menuitem"
          tabindex="-1"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="12"
              cy="7"
              r="4"
            />
          </svg>
          <span>
            Profile
          </span>
          <span
            class="ml-auto text-xs tracking-widest opacity-60"
          >
            ⇧⌘P
          </span>
        </div>
        <div
          class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
          data-orientation="vertical"
          data-radix-collection-item=""
          role="menuitem"
          tabindex="-1"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="14"
              rx="2"
              width="20"
              x="2"
              y="5"
            />
            <line
              x1="2"
              x2="22"
              y1="10"
              y2="10"
            />
          </svg>
          <span>
            Billing
          </span>
          <span
            class="ml-auto text-xs tracking-widest opacity-60"
          >
            ⌘B
          </span>
        </div>
        <div
          class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
          data-orientation="vertical"
          data-radix-collection-item=""
          role="menuitem"
          tabindex="-1"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
          </svg>
          <span>
            Settings
          </span>
          <span
            class="ml-auto text-xs tracking-widest opacity-60"
          >
            ⌘S
          </span>
        </div>
        <div
          class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
          data-orientation="vertical"
          data-radix-collection-item=""
          role="menuitem"
          tabindex="-1"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="16"
              rx="2"
              ry="2"
              width="20"
              x="2"
              y="4"
            />
            <path
              d="M6 8h.001"
            />
            <path
              d="M10 8h.001"
            />
            <path
              d="M14 8h.001"
            />
            <path
              d="M18 8h.001"
            />
            <path
              d="M8 12h.001"
            />
            <path
              d="M12 12h.001"
            />
            <path
              d="M16 12h.001"
            />
            <path
              d="M7 16h10"
            />
          </svg>
          <span>
            Keyboard shortcuts
          </span>
          <span
            class="ml-auto text-xs tracking-widest opacity-60"
          >
            ⌘K
          </span>
        </div>
      </div>
      <div
        aria-orientation="horizontal"
        class="-mx-1 my-1 h-px bg-muted"
        role="separator"
      />
      <div
        role="group"
      >
        <div
          class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
          data-orientation="vertical"
          data-radix-collection-item=""
          role="menuitem"
          tabindex="-1"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <path
              d="M16 3.13a4 4 0 0 1 0 7.75"
            />
          </svg>
          <span>
            Team
          </span>
        </div>
        <div
          aria-controls="radix-:r7:"
          aria-expanded="false"
          aria-haspopup="menu"
          class="flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none focus:bg-accent data-[state=open]:bg-accent"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r8:"
          role="menuitem"
          tabindex="-1"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <line
              x1="19"
              x2="19"
              y1="8"
              y2="14"
            />
            <line
              x1="22"
              x2="16"
              y1="11"
              y2="11"
            />
          </svg>
          <span>
            Invite users
          </span>
          <svg
            class="ml-auto h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="9 18 15 12 9 6"
            />
          </svg>
        </div>
        <div
          class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
          data-orientation="vertical"
          data-radix-collection-item=""
          role="menuitem"
          tabindex="-1"
        >
          <svg
            class="mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
          <span>
            New Team
          </span>
          <span
            class="ml-auto text-xs tracking-widest opacity-60"
          >
            ⌘+T
          </span>
        </div>
      </div>
      <div
        aria-orientation="horizontal"
        class="-mx-1 my-1 h-px bg-muted"
        role="separator"
      />
      <div
        class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <svg
          class="mr-2 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
          />
          <path
            d="M9 18c-4.51 2-5-2-7-2"
          />
        </svg>
        <span>
          GitHub
        </span>
      </div>
      <div
        class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <svg
          class="mr-2 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="m4.93 4.93 4.24 4.24"
          />
          <path
            d="m14.83 9.17 4.24-4.24"
          />
          <path
            d="m14.83 14.83 4.24 4.24"
          />
          <path
            d="m9.17 14.83-4.24 4.24"
          />
          <circle
            cx="12"
            cy="12"
            r="4"
          />
        </svg>
        <span>
          Support
        </span>
      </div>
      <div
        aria-disabled="true"
        class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
        data-disabled=""
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <svg
          class="mr-2 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.5 19H9a7 7 0 1 1 6.71-9h1.79a4.5 4.5 0 1 1 0 9Z"
          />
        </svg>
        <span>
          API
        </span>
      </div>
      <div
        aria-orientation="horizontal"
        class="-mx-1 my-1 h-px bg-muted"
        role="separator"
      />
      <div
        class="relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        <svg
          class="mr-2 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
          />
          <polyline
            points="16 17 21 12 16 7"
          />
          <line
            x1="21"
            x2="9"
            y1="12"
            y2="12"
          />
        </svg>
        <span>
          Log out
        </span>
        <span
          class="ml-auto text-xs tracking-widest opacity-60"
        >
          ⇧⌘Q
        </span>
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
